syntax = "proto3";

package orders.v1;

import "google/protobuf/timestamp.proto";

service OrdersService {
  rpc GetOrdersByUser(GetOrdersByUserRequest) returns (GetOrdersByUserResponse);
}

message GetOrdersByUserRequest {
  int32 user_id = 1;
}

message OrderItem {
  int32 id = 1;
  int32 order_id = 2;
  int32 offer_id = 3;
  int32 quantity = 4;
}

message Order {
  int32 id = 1;
  int32 user_id = 2;

  // nullable in DB
  optional int32 partner_id = 3;

  string order_status = 4;
  string payment_status = 5;

  // nullable in DB
  optional int32 payment_id = 6;

  google.protobuf.Timestamp created_at = 7;
  google.protobuf.Timestamp updated_at = 8;

  repeated OrderItem items = 9;
}

message GetOrdersByUserResponse {
  repeated Order orders = 1;
}
